packages/@sanity/client/src/validators.js

```diff
@@ -2,9 +2,9 @@ const VALID_ASSET_TYPES = ['image', 'file']
 const VALID_INSERT_LOCATIONS = ['before', 'after', 'replace']
 
 exports.dataset = name => {
-  if (!/^[-\w]{1,128}$/.test(name)) {
+  if (!/^[~][a-z0-9][-\w]{2,20}$|^[a-z0-9][-\w]{1,20}$/.test(name)) {
```

Comment:
The regex is a bit off though - aliases (should) need between 1 and 26 characters, where the leading character cannot be a dash, eg `^~[a-z0-9]{1}[-\w]{0,25}$`. Datasets are between 1 and 20 characters, eg `^[a-z0-9][-\w]{0,19}`. I believe this catch both cases:

```suggestion
  if (!/^(~[a-z0-9]{1}[-\w]{0,25}|[a-z0-9]{1}[-\w]{0,19})$/.test(name)) {
```